invoices/invoices.go

```diff
@@ -471,6 +471,20 @@ func (i *Invoice) HTLCSetCompliment(setID *[32]byte,
 	return htlcSet
 }
 
+// IsKeysend returns true if the invoice is a Keysend invoice.
+func (i *Invoice) IsKeysend() bool {
+	// TODO(positiveblue): look for a more reliable way to tests if
+	// an invoice is keysend.
+	return len(i.PaymentRequest) == 0 && !i.IsAMP()
+}
+
+// IsAMP returns true if the invoice is an AMP invoice.
+func (i *Invoice) IsAMP() bool {
+	return i.Terms.Features.HasFeature(
```

Comment:
Do we need to check if `Features` is `nil` here and return false? Since this might now be used/called in situations where there aren't any features defined.

Reply:
:dart: